File: map_reduce.py
from bson.code import Code
from pymongo import MongoClient

#connect to database
client = MongoClient()
db = client.seances

map_function = """
function() {
    emit(this.Film.Titre, this.Entree);
}
"""

reduce_function = """
function(key, values) {
    var sum = 0;
    for (var i = 0; i < values.length; i++) {
        sum += values[i];
    }
    return sum / values.length;
}
"""

result = db.seances.map_reduce(map_function, reduce_function, "average_entree_by_film", query={}, out={"replace": "results"})

for doc in result.find():
    print(doc)

File: map_reduce.js
//mapReduce deprecated after version 5.0 of mongoDB

var mapFunction = function() {
    emit(this.Filme.Titre, { totalEntree: this.Entree, count: 1 });
};

var reduceFunction = function(key, values) {
    var reducedObject = { totalEntree: 0, count: 0 };

    for (var i = 0; i < values.length; i++) {
        reducedObject.totalEntree += values[i].totalEntree;
        reducedObject.count += values[i].count;
    }

    return reducedObject;
};

var finalizeFunction = function(key, reducedValue) {
    reducedValue.avgEntree = reducedValue.totalEntree / reducedValue.count;
    return reducedValue;
};

db.seances.mapReduce(
    mapFunction,
    reduceFunction,
    {
        out: "film_entree_average",
        finalize: finalizeFunction
    }
);